tests/driver_ws281x/Makefile

```diff
@@ -1,4 +1,4 @@
-BOARD ?= atmega328p
+BOARD ?= native
```

Comment:
How about adding:

```Makefile
ifneq (native,$(BOARD))
  FEATURES_REQUIRED := arch_avr8
endif
```

Reply:
That's what I tried [first](https://github.com/RIOT-OS/RIOT/pull/12793/commits/ed08dd5f9396d666689b1ca5504351dd2e19d720#diff-e7b9245157e79b99a03a00b1da32af85L10).

The problem is that Murdock then parsed the file differently than `make` would (not sure which script is responsible for it) and ended up building it for all platforms, ignoring the conditional `FEATURES_REQUIRED`.

Reply:
Murdock uses `make info-boards-supported` to get the list of supported boards. In the worst case, we would need to add `FEATURES_PROVIDED += ws281x_backend` to `cpu/atmega_common/Makefile.features` and to `cpu/native/Makefile.features` and require that... But I'm not really thrilled but that idea.

Reply:
> The problem is that Murdock then parsed the file differently than `make` would (not sure which script is responsible for it) and ended up building it for all platforms, ignoring the conditional `FEATURES_REQUIRED`.

No, the issue is that the main application Makefile always gets fully evaluated *before* info-boards-supported, independent of Murdock.

So the `if native: require avr` thing needs to be somewhere in Makefile.dep. (It seems weird to begin with, though though. :wink: )

Reply:
> So the `if native: require avr` thing needs to be somewhere in Makefile.dep. (It seems weird to begin with, though though. )

That will work for native, but what if we add that SPI backed?
Testing the contents of `FEATURES_PROVIDED` is not allowed and I'd rather not maintain a `BOARD_PROVIDES_SPI` list *shiver*